webapp/src/components/github_repo_selector/github_repo_selector.jsx

```diff
@@ -30,25 +33,65 @@ export default class GithubRepoSelector extends PureComponent {
     }
 
     componentDidMount() {
-        this.props.actions.getRepos();
+        this.props.actions.getOrgs();
+        this.props.actions.getReposByOrg("");
     }
 
-    onChange = (_, name) => {
-        const repo = this.props.yourRepos.find((r) => r.full_name === name);
+    onChangeForOrg = (_, org) => {
+        if (this.state.org != org) {
+            this.setState({org: org}, () => {this.props.actions.getReposByOrg(org)});
+            this.props.onChange(null);
+        }
+    }
+
+    onChangeForRepo = (_, name) => {
+        const repo = this.props.yourReposByOrg.find((r) => r.full_name === name);
         this.props.onChange({name, permissions: repo.permissions});
     }
 
     render() {
-        const repoOptions = this.props.yourRepos.map((item) => ({value: item.full_name, label: item.full_name}));
+        const orgOptions  = this.props.yourOrgs.map((item) => ({value: item.login, label: item.login}));
+        orgOptions.unshift({value: "", label: "your owned repositories"});
+        const repoOptions = this.props.yourReposByOrg.map((item) => ({value: item.full_name, label: item.name}));
+        let orgSelector, helperTextForRepoSelector;
```

Comment:
```suggestion
        let orgSelector = null;
        let helperTextForRepoSelector = 'Returns GitHub repositories connected to the user account';
```

Then we can remove the `else` block below

Reply:
Done.